src/embedded/embeddedSessionManager.ts

```diff
@@ -1,52 +1,48 @@
 import { v4 as uuidv4 } from 'uuid';
-import {
-  IEmbeddedImpressionData,
-  IEmbeddedSession
-} from '../../src/events/embedded/types';
 import { trackEmbeddedSession } from '../events/embedded/events';
+import { IterableEmbeddedSessionRequestPayload } from '..';
 
 class EmbeddedSession {
   public start?: Date;
   public end?: Date;
-  public placementId?: string;
-  public impressions?: Array<IEmbeddedImpressionData>;
+  public impressions?: EmbeddedImpression[];
   public id: string;
 
-  constructor(
-    start?: Date,
-    end?: Date,
-    placementId?: string,
-    impressions?: Array<IEmbeddedImpressionData>
-  ) {
+  constructor(start?: Date, end?: Date) {
     this.start = start;
     this.end = end;
-    this.placementId = placementId;
-    this.impressions = impressions;
     this.id = uuidv4();
   }
 }
 
-class EmbeddedImpressionData {
+class EmbeddedImpression {
   public messageId: string;
   public displayCount: number;
-  public duration: number;
+  public displayDuration: number;
   public start?: Date = undefined;
+  public placementId: number;
 
-  constructor(messageId: string, displayCount?: number, duration?: number) {
+  constructor(
+    messageId: string,
+    placementId: number,
+    displayCount?: number,
+    duration?: number
+  ) {
     this.messageId = messageId;
     this.displayCount = displayCount ? displayCount : 0;
-    this.duration = duration ? duration : 0.0;
+    this.displayDuration = duration ? duration : 0.0;
+    this.placementId = placementId;
   }
 }
 
-export class EmbeddedSessionManager {
-  private impressions: Map<string, IEmbeddedImpressionData> = new Map();
-  public session: IEmbeddedSession = new EmbeddedSession(
-    undefined,
-    undefined,
-    '0',
-    undefined
-  );
+export class IterableEmbeddedSessionManager {
+  public appPackageName: string;
+  private impressions: Map<string, EmbeddedImpression> = new Map();
+  public session: EmbeddedSession = new EmbeddedSession();
```

Comment:
@mprew97 Something that trips me up here — it feels to me like the impressions should be part of the session, since that's how we track them in the API

Reply:
Either that, or... why do we need a session object? If a session's impressions are being tracked at the top level of the session manager... should everything else about a session be tracked that way, too?

Reply:
@hardikmashru had replied on a separate PR that they were following the Android convention. Happy for us to swap over if we think it's necessary. 

and tbf, the request body of `embedded/session` has `impressions` and `session` as siblings for the parameters... likely why they are separated.

Reply:
@mprew97 OK, spent some time looking at the API and the code, and I follow what you're saying. Thanks!